### ISSUE TITLE
OSX agent fails on long running builds

### ISSUE THREAD

**Issue by simmdan** [Microsoft/vsts-agent#55]
I have a build configuration which downloads binaries created by another build and then runs a series of very timeconsuming tests (they take 2+ hours to complete on a linux VM).  If I run the tests on OSX using the vsts-agent, they fail after about 15 minutes with a message like this:

> The job has been abandoned because agent prose-mac-test-1 did not renew the lock. Ensure agent is running, not sleeping, and has not lost communication with the service.

When this first happens, no log information is available from the main task that runs the tests and the build agent shows a red bar even though it says that it's status is idle.  After another 5 minutes or so, the job finally completes and the agent status goes back to green.

Possibly relevant is that all of my tests are run by executing a single executable (a custom test runner) configured as a "CommandLine" task in the build.  This executable generates lots of output to stdio which does appear in the console during the first 15 minutes before the error comes up.

Also note that essentially the exact same configuration runs to completion over 2+ hours on an ubuntu linux machine.

**Comment by simmdan** [comment]
Interestingly, I just setup the older vso-agent, and I'm getting the same results with it.

**Comment by bryanmacfarlane** [comment]
Interesting.  Is it always around the same time?  the agent should renew a lock essentially saying I'm still alive - every 30 seconds or so and the server would have to not hear from it for 5+ min (10 missed renewals) before the server kills the job.  

I wonder if the same happens for you if you ran tasks which just slept and output every 10 seconds in a loop for 30 minutes ...

**Comment by simmdan** [comment]
Pretty sure that this was my fault.  I tried with vso-agent with the same results.  Then this morning I stumbled on the fact that the default power settings for my mac mini were to go to sleep if the monitor is turned off, and since I spend most of the time with my kvm set to point to another machine and just use ssh to get to the mac, it looks like it was powering down in the middle of the test runs or something like that.

Now that I have adjusted the power settings, I'm currently 43.5 minutes into a run and everything is still going strong.  I will update with more info as my tests continue, but probably this issue should be closed.
